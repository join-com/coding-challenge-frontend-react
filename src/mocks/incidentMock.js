const incidentMock = {
  id: 24134,
  title: "Stolen di Florino (red and white)",
  description: "",
  address: "Berlin, 10245, DE",
  occurred_at: 1402984800,
  updated_at: 1545551568,
  url: "https://bikewise.org/api/v1/incidents/24134",
  source: {
    name: "BikeIndex.org",
    html_url: "https://bikeindex.org/bikes/27159",
    api_url: "https://bikeindex.org/api/v1/bikes/27159"
  },
  media: {
    image_url: "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/large_dsc_1293.jpg",
    image_url_thumb: "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/small_dsc_1293.jpg"
  },
  location_type: null,
  location_description: null,
  type: "Theft",
  type_properties: null
};

export default incidentMock;
