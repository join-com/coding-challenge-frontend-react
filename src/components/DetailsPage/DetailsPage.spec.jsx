import React from "react";
import { shallow } from "enzyme";
import Component from "./DetailsPage";
import incidentMock from "../../mocks/incidentMock";

describe("DetailsPage component test", () => {
  const defaultProps = {
    fetchIncidentById: jest.fn(),
    match: {
      params: {
        id: incidentMock.id
      }
    },
    item: incidentMock,
    isLoading: false,
    error: null
  };

  const getWrapper = (testProps = {}) => {
    const props = { ...defaultProps, ...testProps };
    return shallow(<Component {...props} />);
  };

  it("should match snapshot and call fetchIncidentById prop function", () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
    expect(defaultProps.fetchIncidentById).toHaveBeenCalledTimes(1);
  });

  it("should match snapshot for loading state", () => {
    const wrapper = getWrapper({ isLoading: true, item: {} });
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot for loading state", () => {
    const wrapper = getWrapper({ error: "error message" });
    expect(wrapper).toMatchSnapshot();
  });
});
