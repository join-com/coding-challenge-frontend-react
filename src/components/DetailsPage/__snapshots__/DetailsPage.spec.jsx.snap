// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage component test should match snapshot and call fetchIncidentById prop function 1`] = `
<styled.div>
  <styled.div>
    <Heading
      bold={false}
      level={3}
      uppercase={false}
    >
      Stolen di Florino (red and white)
    </Heading>
    <Text
      bold={false}
      size="normal"
      uppercase={false}
    >
      <b>
        Stolen
      </b>
       
      Di., 17. Juni 2014
       
      <b>
        at
      </b>
       
      Berlin, 10245, DE
    </Text>
  </styled.div>
</styled.div>
`;

exports[`DetailsPage component test should match snapshot for loading state 1`] = `
<styled.div>
  <Text
    bold={false}
    size="normal"
    uppercase={false}
  >
    Loading ...
  </Text>
</styled.div>
`;

exports[`DetailsPage component test should match snapshot for loading state 2`] = `
<styled.div>
  <ErrorMessage
    message="error message"
  />
</styled.div>
`;
